test/extended/networking/networkpolicy.go

```diff
@@ -363,7 +392,38 @@ func testCannotConnect(f *framework.Framework, ns *kapiv1.Namespace, podName str
 
 	framework.Logf("Waiting for %s to complete.", podClient.Name)
 	err := framework.WaitForPodSuccessInNamespace(f.ClientSet, podClient.Name, ns.Name)
-	Expect(err).To(HaveOccurred(), fmt.Sprintf("checking %s could not communicate with server.", podName))
+
+	// We expect an error here since it's a cannot connect test.
+	// Dump debug information if the error was nil.
+	if err == nil {
+		// Collect pod logs when we see a failure.
+		logs, logErr := framework.GetPodLogs(f.ClientSet, f.Namespace.Name, podName, fmt.Sprintf("%s-container", podName))
```

Comment:
Instead of repeating same thing in https://github.com/danwinship/origin/blob/81d61cdcf9b30a510d345599e1c5ef0db680d09b/test/extended/networking/networkpolicy.go#L353-L379
May be create a wrapper method dumpDebugInfo() or sth and reuse in both places?

Reply:
It's just copied straight from upstream. I wanted to stay in sync with them for the changes I copied over, to make it easier to copy other stuff over in the future